### PR TITLE
Update qmd-progress package.py

### PR DESCRIPTION
**Pull request by jeanlucf22**

### CODE REVIEW

File: var/spack/repos/builtin/packages/qmd-progress/package.py [hunk 1]
```diff
@@ -12,17 +12,21 @@ class QmdProgress(CMakePackage):
...
+    url = "https://github.com/lanl/qmd-progress/archive/refs/tags/v1.2.0.tar.gz"
     git = "https://github.com/lanl/qmd-progress.git"
 
-    version("develop", branch="master")
+    maintainers = ["jeanlucf22"]
+
+    version("master", branch="master")
+    version("1.2.0", sha256="d41708f0e9c12d0e421a9fa883f8b12478cf4faa7453703143f32c093626638e")
     version("1.1.0", sha256="2c5eac252067bfb55d715c9ce5de2e4306b20b4273979dda15b4a2f71f69bb0b")
     version("1.0.0", sha256="28c99eb80d9a6b09e1d01d61538b3b924850d89c6a8bfb5d3e8b6490be822296")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/qmd-progress/package.py]: The sha256 for these versions are no longer valid.  Any idea why the archive version doesn't match the tarball version?

```
$ spack checksum qmd-progress 1.1.0 1.0.0
==> Found 2 versions of qmd-progress:
  
  1.1.0  https://github.com/lanl/qmd-progress/archive/refs/tags/v1.1.0.tar.gz
  1.0.0  https://github.com/lanl/qmd-progress/archive/refs/tags/v1.0.0.tar.gz

==> Fetching https://github.com/lanl/qmd-progress/archive/refs/tags/v1.1.0.tar.gz
==> Fetching https://github.com/lanl/qmd-progress/archive/refs/tags/v1.0.0.tar.gz

    version("1.1.0", sha256="757d2606d7b7f38e7f8f491bf7369b88de55062bae0b12a9928f0a5acae993bd")
    version("1.0.0", sha256="eed87e68b4a1533a3ed70c1662feca4ac890b985f3326fc94522c2f71f198fdc")
```

You could add `url_for_version` for those at or below `1.1.0` (see https://spack.readthedocs.io/en/latest/packaging_guide.html#version-urls) or update them.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the homepage, branch, and new version; however, the old version hashes no longer match with the new url.

File: var/spack/repos/builtin/packages/qmd-progress/package.py (hunk 1 above)
Comment (jeanlucf22) [reply var/spack/repos/builtin/packages/qmd-progress/package.py]: I updated the old sha256

**Review by tldahlgren — APPROVED** [review]
Confirmed the branch (existence) and the three sha256.

**Pull request merged**